src/sentry/tasks/base.py

```diff
@@ -23,8 +23,14 @@ def _wrapped(*args, **kwargs):
                 instance = '{}.{}'.format(name, stat_suffix(*args, **kwargs))
             else:
                 instance = name
-            with metrics.timer(key, instance=instance):
-                result = func(*args, **kwargs)
+            with metrics.timer(key, instance=instance) as tags:
+                try:
+                    result = func(*args, **kwargs)
+                except Exception:
+                    tags['result'] = 'failure'
+                    raise
```

Comment:
Should we use `six.reraise`? Going to look at what that does, exactly. Since we use it in other places.